ragna/core/_rag.py

```diff
@@ -188,7 +196,17 @@ async def prepare(self) -> Message:
                 detail=RagnaException.EVENT,
             )
 
-        await self._run(self.source_storage.store, self.documents)
+        from ragna.core import Document
+        if type(self.source_storage).__ragna_input_type__ == Document:
+            await self._run(self.source_storage.store, self.documents)
+        else:
+            # Here we need to generate the list of embeddings
+            chunks = []
+            for document in self.documents:
+                chunks += self.source_storage._chunk_pages(document.extract_pages(), document_id=document.id, chunk_size=500, chunk_overlap=250)
```

Comment:
This logic needs to be moved to the `MiniLML6v2` class. We can no longer put this on the source storage.